### PR TITLE
Minor tweaks from experimenting

### PR DESCRIPTION
**Pull request by Kha**

### CODE REVIEW

File: src/Init/Lean/Parser/Syntax.lean [hunk 1]
```diff
@@ -32,10 +32,10 @@ namespace Syntax
 @[builtinSyntaxParser] def ident     := parser! nonReservedSymbol "ident"
 @[builtinSyntaxParser] def try       := parser! nonReservedSymbol "try " >> syntaxParser
 @[builtinSyntaxParser] def lookahead := parser! nonReservedSymbol "lookahead " >> syntaxParser
-@[builtinSyntaxParser] def optional  := parser! nonReservedSymbol "optional " >> syntaxParser
 @[builtinSyntaxParser] def sepBy     := parser! nonReservedSymbol "sepBy " >> syntaxParser >> syntaxParser
 @[builtinSyntaxParser] def sepBy1    := parser! nonReservedSymbol "sepBy1 " >> syntaxParser >> syntaxParser
 
+@[builtinSyntaxParser] def optional  := tparser! pushLeading >> symbolAux "?" none
```
Comment (Kha) [inline src/Init/Lean/Parser/Syntax.lean]: @leodemoura I realized only after the fact that `p?` parses as a single identifier D: ... that is pretty unfortunate. I would still prefer `(p)?` over `(optional p)`, I think.

Comment (leodemoura) [reply src/Init/Lean/Parser/Syntax.lean]: I got disappointed when I tried it too. It is the price of being able to write `x.find?` :(
BTW, I will change the precedence of `try`, `sepBy` nested parsers.
Right now, `try p1 p2` parsers as `try (p1 p2)`.

File: src/Init/Lean/Parser/Parser.lean [hunk 2]
```diff
@@ -1484,7 +1492,8 @@ let nameP := checkNoWsBefore ("no space before ':" ++ name ++ "'") >> symbolAux
 -- if parsing the kind fails and `anonymous` is true, check that we're not ignoring a different
 -- antiquotation kind via `noImmediateColon`
 let nameP := if anonymous then nameP <|> noImmediateColon >> pushNone >> pushNone else nameP;
-node kind $ try $ dollarSymbol >> checkNoWsBefore "no space before" >> antiquotExpr >> nameP >> optional (checkNoWsBefore "" >> "*")
+-- antiquotations are not part of the "standard" syntax, so hide "expected '$'" on error
+node kind $ try $ setExpected [] dollarSymbol >> checkNoWsBefore "no space before" >> antiquotExpr >> nameP >> optional (checkNoWsBefore "" >> "*")
```
Comment (leodemoura) [inline src/Init/Lean/Parser/Parser.lean]: This is a great change. The expected `$` in error messages was really weird :)

**Comment by leodemoura** [comment]
merged

**Pull request closed without merging**